src/jdk.incubator.jextract/share/classes/jdk/incubator/jextract/Declaration.java

```diff
@@ -54,6 +55,19 @@
      */
     String name();
 
+    /**
+     * The values of the specified attribute.
+     * @param name The attribute to retrieve
+     * @return The list of values associate with this attribute
+     */
+    Optional<List<ConstantDesc>> getAttribute(String name);
+
+    /**
+     * The attributes associated with this declaration
+     * @return The attributes associated with this declaration
+     */
+    Set<String> attributeNames();
```

Comment:
I find it odd that the API has ways to query attributes, but no way to set them. I really think that withAttribute/dropAttributes should be a pretty minimal way to get there.